### PR TITLE
add pdf , html render, add test data generator, remove dicts when pos…

### PR DESCRIPTION
**Pull request by nichtsfrei**
…sible

add rudimentary pdf and html renderer to verify data usage used in
views.

add test data generator to quickly produce test xmls with n hosts and n
oids per host.

remove dicts when possible to reduce memory consumption but higher cpu usage.

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/pheme/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: pheme/transformation/scanreport/gvmd.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 from functools import reduce
 from typing import Callable, Optional, Dict
+import logging
+
+# from guppy import hpy
```
Comment (bjoernricks) [inline pheme/transformation/scanreport/gvmd.py:23]:
```suggestion
```

File: pheme/transformation/scanreport/gvmd.py [hunk 2]
```diff
@@ -123,22 +127,38 @@ def group_by_nvt(first: Dict[str, NVTResult], second: Dict[str, str]):
...
 
+logger = logging.getLogger(__name__)
+
+
 def transform(
     data: Dict[str, str], group_by: Callable = group_by_host
 ) -> Report:
-    report = data["report"]["report"]
+    #    h = hpy()
+    #    h.setref()
```
Comment (bjoernricks) [inline pheme/transformation/scanreport/gvmd.py:142]:
```suggestion
```

File: pheme/transformation/scanreport/gvmd.py [hunk 3]
```diff
@@ -201,13 +222,15 @@ def may_create_result_count() -> ResultCount:
...
+            original_results.get('start'),
             list(grouped.values()),
         ),
         may_create_filtered(report, 'severity'),
         may_create_result_count(),
     )
     del grouped
+    del original_results
     del report
+    #    logger.debug(h.heap())
```
Comment (bjoernricks) [inline pheme/transformation/scanreport/gvmd.py:235]:
```suggestion
```

File: poetry.lock [hunk 4]
```diff
@@ -242,6 +242,14 @@ djangorestframework = ">=3.9"
...
 version = ">=3.7.4"
 
+[[package]]
+category = "main"
+description = "Guppy 3 -- Guppy-PE ported to Python 3"
+name = "guppy3"
+optional = false
+python-versions = ">=3.5"
+version = "3.0.10.post1"
+
```
Comment (bjoernricks) [inline poetry.lock:252]: Do you use guppy3 actually?

File: pyproject.toml [hunk 5]
```diff
@@ -32,6 +32,7 @@ coreapi = "^2.3.3"
 pytest = "^6.0.1"
 pytest-cov = "^2.10.0"
 pytest-django = "^3.9.0"
+guppy3 = "^3.0.10"
```
Comment (bjoernricks) [inline pyproject.toml:35]: Same here. I can't find an import for guppy3

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: poetry.lock (hunk 4 above)
Comment (nichtsfrei) [reply poetry.lock:252]: I used it to debug the heap, but you're right, it doesn't need to stay permanently in the project. Going to remove.

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**